File: qoj.ac/2114/L.cpp
#include <bits/stdc++.h>
#define int long long

// 1 4 21 120 697 4060

void solve() {
  
}

int32_t main() {
  std::cin.tie(nullptr)->sync_with_stdio(false);

  std::vector<int> f(101);
  f[1] = 3, f[2] = 20;
  for (int i = 3; i <= 100; i++) {
    f[i] = 6 * f[i - 1] - f[i - 2] + 2;
  }
  for (int i = 1; i <= 100; i++) {
    std::cerr << i << ": " << f[i] << '\n';
  }

  int prv = 1;
  for (int L = 1; L <= 4060; L++) {
    for (int n = L;; n++) {
      int l = 1, r = n, ans = -1;
      while (l <= r) {
        int mid = (l + r) / 2;
        if (mid * (mid + 1) * 2 <= n * (n + 1)) l = mid + 1, ans = mid;
        else r = mid - 1;
      }
      if (ans * (ans + 1) * 2 == n * (n + 1)) {
        if (n != prv) {
          std::cout << n << ' ' << L << ' ' << n - 6 * prv << '\n';
          prv = n;
        }
        break;
      }
    }
  }

  int t = 1;
  std::cin >> t;

  while (t--) {
    solve();
  }

  return 0;
}

File: qoj.ac/2114/L.py
T = int(input())

mx = 1e200
f = [0, 3, 20]

len = 2
for i in range(3, 200000):
  f.append(6 * f[i - 1] - f[i - 2] + 2)
  len += 1
  if f[-1] > mx:
    break

# for i in f:
#   print(i)

for _ in range(T):
  n = int(input())
  l, r = 1, len
  ans = -1
  while l <= r:
    mid = (l + r) // 2
    if f[mid] > n:
      ans = f[mid]
      r = mid - 1
    else:
      l = mid + 1
  print(ans)